crates/sargon/src/system/drivers/drivers.rs

```diff
@@ -42,6 +42,14 @@ impl Drivers {
 
 #[cfg(test)]
 impl Drivers {
+    fn file_system() -> Arc<dyn FileSystemDriver> {
```

Comment:
This allows for easy testing, it is very inconvenient to use real file system operation for unit tests, it makes them stateful! instead for unit tests we wanna use an in memory file system